src/toil/jobStores/azureJobStore.py

```diff
@@ -800,12 +842,27 @@ def attempt(last=False):
         else:
             go.pop()
 
+    total_tries = num_tries
+    totalDelay = 0
+    delay = 0
     while go:
-        if num_tries == 1:
```

Comment:
You should guarantee that at least one attempt is being made, even if caller fudges up the timeout and delays arguments. That or raise an exception. The current code will just skip the attempt, right?